app/models/concerns/offer_search.rb

```diff
@@ -3,36 +3,34 @@ module OfferSearch
   extend ActiveSupport::Concern
 
   included do
-
     # offer: notes
     # user: first_name, last_name, email, mobile
     # gogovan: driver name, driver rego, driver number
     # items: donor_description
     # messages.body
     # package_type: name_en, name_zh_tw
-
-    scope :search, -> (options = {}) {
+    scope :search, lambda { |options = {}|
       search_text = options[:search_text] || ''
-      search_query = ['offers.notes', 'offers.id', 'users.first_name', 'users.last_name',
-         'users.email', 'users.mobile', 'items.donor_description',
-         'messages.body', 'companies.name', 'packages.inventory_number',
-         'package_types.name_en', 'package_types.name_zh_tw',
-         'gogovan_orders.driver_name', 'gogovan_orders.driver_mobile', 'gogovan_orders.driver_license'
-        ].
-        map { |f| "#{f}::text ILIKE :search_text" }.
-        join(" OR ")
+      search_query = ['offers.notes', 'offers.id', 'users.first_name',
+                      'users.last_name', 'users.email', 'users.mobile',
+                      'items.donor_description', 'messages.body',
+                      'companies.name', 'packages.inventory_number',
+                      'package_types.name_en', 'package_types.name_zh_tw',
+                      'gogovan_orders.driver_name',
+                      'gogovan_orders.driver_mobile',
+                      'gogovan_orders.driver_license']
+                     .map { |f| "#{f}::text ILIKE :search_text" }
+                     .join(' OR ')
       where(search_query, search_text: "%#{search_text}%")
-        .joins("LEFT OUTER JOIN companies ON offers.company_id = companies.id")
-        .joins("LEFT OUTER JOIN users ON offers.created_by_id = users.id")
-        .joins("LEFT OUTER JOIN items ON offers.id = items.offer_id")
-        .joins("LEFT OUTER JOIN messages ON offers.id = messages.offer_id OR items.id = messages.item_id")
-        .joins("LEFT OUTER JOIN packages ON packages.item_id = items.id")
-        .joins("LEFT OUTER JOIN package_types ON package_types.id = packages.package_type_id")
-        .joins("LEFT OUTER JOIN deliveries ON offers.id = deliveries.offer_id")
-        .joins("LEFT OUTER JOIN gogovan_orders ON gogovan_orders.id = deliveries.gogovan_order_id")
+        .joins('LEFT OUTER JOIN companies ON offers.company_id = companies.id')
+        .joins('LEFT OUTER JOIN users ON offers.created_by_id = users.id')
+        .joins('LEFT OUTER JOIN items ON offers.id = items.offer_id')
+        .joins("LEFT OUTER JOIN messages ON offers.id = messages.messageable_id and messages.messageable_type = 'Offer'")
```

Comment:
Don't forget we need messages on items too. You can reference `items.id` as this is already joined above. 

```suggestion
        .joins("LEFT OUTER JOIN messages ON offers.id = messages.messageable_id and messages.messageable_type = 'Offer'")
        .joins("LEFT OUTER JOIN messages ON items.id = messages.messageable_id and messages.messageable_type = 'Item'")
```

Make sure there are some specs to confirm this works and doesn't load items from other unrelated offers.